blockprod/src/detail/mod.rs

```diff
@@ -141,14 +149,70 @@ impl BlockProduction {
                         })
                     };
 
+                    let block_height = best_block_index.block_height().next_height();
+                    let sealed_epoch_index = chain_config.sealed_epoch_index(&block_height);
+
+                    let sealed_epoch_randomness = sealed_epoch_index
+                        .map(|index| this.get_epoch_data(index))
+                        .transpose()
+                        .map_err(|_| {
+                            ConsensusPoSError::PropertyQueryError(
+                                PropertyQueryError::EpochDataNotFound,
+                            )
+                        })?
+                        .flatten()
+                        .or_else(|| Some(EpochData::new(PoSRandomness::at_genesis(&chain_config))));
```

Comment:
I'd place a todo here, we have the same one in `consensus/src/pos/mod.rs`

Reply:
I think there's a few places where `get_epoch_data()` is called, and if no data is found then we default to `PoSRandomness::at_genesis()` rather than error. Maybe this has changed, or it's time to change this in all occurrences?

Reply:
yes, we default to genesis but this is only ok unless we seal an epoch, after that it should be an error. Right now there are todos for that all over the code so I think it should be placed here as well.

Reply:
TODO added.